packages/page-staking/src/useAbftScores.tsx

```diff
@@ -0,0 +1,78 @@
+// Copyright 2017-2025 @polkadot/app-staking authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import type { ApiPromise } from '@polkadot/api';
+import type { AbftScores } from './types.js';
+
+import { useEffect, useState } from 'react';
+
+import { getCommitteeManagement } from '@polkadot/react-api/getCommitteeManagement';
+import { createNamedHook, useApi } from '@polkadot/react-hooks';
+
+import useSessionInfo from './Performance/useSessionInfo.js';
+import { getSessionFirstAndLastBlock } from './Query/util.js';
+import { decodeChainAbftScore } from './util.js';
+
+/**
+ * Retrieving ABFT scores from the storage for the past sessions, and from the current session.
+ * For the past sessions, it retrieves, in a sort of pipeline fashion, below data, for each session passed as input
+ *    - last session block,
+ *    - api from that block,
+ *    - api.query.aleph.abftScores(session)
+ *  This logic might seem complicated  - why not just call api.query.aleph.abftScores(somePastSession) via current API?
+ *  Because in each session % 960 == 0, the ABFT score storage is cleared, and validator needs to access past score data
+ *  regardless.
+ *
+ *  For the current session, it uses the current api.query.aleph.abftScores.
+ * @param sessions An array of session numbers. Only past or current session numbers can be called.
+ */
+function useAbftScoresImpl (sessions: number[]): AbftScores[] {
+  const { api } = useApi();
+
+  const sessionInfo = useSessionInfo();
+  const currentSession = sessionInfo?.currentSession;
+
+  const [abftScores, setAbftScores] = useState<AbftScores[]>([]);
+
+  const fetchAbtfScoreForSession = async (sessionNo: number, api: ApiPromise, currentSessionNo: number) => {
+    if (sessionNo === currentSessionNo) {
+      return api.query.aleph.abftScores(currentSessionNo);
+    }
+
+    const sessionPeriod = Number(getCommitteeManagement(api).consts.sessionPeriod.toString());
+    const lastSessionBlockNo = getSessionFirstAndLastBlock(sessionNo, sessionPeriod).last;
+    const hash = await api.rpc.chain.getBlockHash(lastSessionBlockNo);
+
+    if (hash.isEmpty) {
+      return undefined;
+    }
+
+    const apiAtBlock = await api.at(hash.toString());
+
+    return apiAtBlock.query.aleph.abftScores ? apiAtBlock.query.aleph.abftScores(sessionNo) : undefined;
+  };
```

Comment:
I'd move this function outside hook to improve readability and not recreate it

Reply:
Sure